script/windows-installer/inno-setup-git-lfs-installer.iss

```diff
@@ -101,6 +101,56 @@ begin
     Result := Pos(';' + UpperCase(ParamExpanded) + '\;', ';' + UpperCase(OrigPath) + ';') = 0;
 end;
 
+function SetEnvironmentVariable(lpName,lpValue:String):Boolean;
+external 'SetEnvironmentVariableW@Kernel32.dll stdcall delayload';
+
+// When Git for Windows is installed with the PATH option "Bash only", i.e.
+// _without_ adding anything to the global `PATH`, we will not find `git.exe`
+// there.
+//
+// Detect that situation and add `<Git>\cmd` to the `PATH` so that we find it
+// when registering Git LFS later.
+function AddGitForWindowsCMDToPATHIfNeeded: boolean;
+var
+  Domain: Integer;
+  Key, PathOption, AppPath, Path: string;
+begin
+  Result := False;
+
+  Key := 'Microsoft\Windows\CurrentVersion\Uninstall\Git_is1';
+  if RegKeyExists(HKEY_LOCAL_MACHINE, 'Software\Wow6432Node\' + Key) then begin
+    Domain := HKEY_LOCAL_MACHINE;
+    Key := 'Software\Wow6432Node\' + Key;
+  end else if RegKeyExists(HKEY_CURRENT_USER, 'Software\Wow6432Node\' + Key) then begin
+    Domain := HKEY_CURRENT_USER;
+    Key := 'Software\Wow6432Node\' + Key;
+  end else if RegKeyExists(HKEY_LOCAL_MACHINE, 'Software\' + Key) then begin
+    Domain := HKEY_LOCAL_MACHINE;
+    Key := 'Software\' + Key;
+  end else if RegKeyExists(HKEY_CURRENT_USER, 'Software\' + Key) then begin
+    Domain := HKEY_CURRENT_USER;
+    Key := 'Software\' + Key;
+  end else
+    Exit;
+
+  if (not RegQueryStringValue(Domain, Key, 'Inno Setup CodeFile: Path Option', PathOption)) or
+    (PathOption <> 'Cmd') or // (PathOption <> 'BashOnly') or
+    (not RegQueryStringValue(Domain, Key, 'Inno Setup: App Path', AppPath)) or
+    (not FileExists(AppPath + '\cmd\git.exe'))
+  then
+    Exit;
+
+  // Extend PATH so that it finds `git.exe`
+  Path := GetEnv('PATH');
+  if Path<>''
+  then begin
+    Path := AppPath + '\cmd;' + Path;
+  end;
```

Comment:
I realize it's very unlikely that `Path` would be empty, but since we're checking the non-empty case, should we include an else block like:
```suggestion
  if Path = '' then
    Path := AppPath + '\cmd'
  else
    Path := AppPath + '\cmd;' + Path;
```